include/deal.II/lac/trilinos_tpetra_sparse_matrix.templates.h

```diff
@@ -0,0 +1,661 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_trilinos_tpetra_sparse_matrix_templates_h
+#define dealii_trilinos_tpetra_sparse_matrix_templates_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+
+#  include <deal.II/lac/dynamic_sparsity_pattern.h>
+#  include <deal.II/lac/trilinos_tpetra_sparse_matrix.h>
+#  include <deal.II/lac/trilinos_tpetra_sparsity_pattern.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+
+  namespace TpetraWrappers
+  {
+    // reinit_matrix():
+    namespace
+    {
+      using size_type = dealii::types::signed_global_dof_index;
+
+      template <typename Node>
+      using MapType =
+        Tpetra::Map<int, dealii::types::signed_global_dof_index, Node>;
+
+      template <typename Number, typename Node>
+      using MatrixType = Tpetra::
+        CrsMatrix<Number, int, dealii::types::signed_global_dof_index, Node>;
+
+      template <typename Node>
+      using GraphType =
+        Tpetra::CrsGraph<int, dealii::types::signed_global_dof_index, Node>;
```

Comment:
`MatrixType`  depends on the scalar type, and these are all strictly speaking copies of what is used inside the `SparseMatrix` class. Could we avoid making these `using` declarations at the global level? I'll write a patch for the existing class as well.

Reply:
I am not sure what I should change here.